### PR TITLE
INFINITY-2031 User-facing changes to 'endpoints' API

### PR DESCRIPTION
**Pull request by nickbp**
This covers changes that affect the EndpointsResource command. Changes for the yaml spec will be done in a separate PR.

- Files VIPs against port name rather than VIP name
- Removes 'vips' list removed due to the above
- Adds validation that port names (which are used as endpoint names across the service) cannot be duplicated. This validation will later be loosed to only apply when a port is configured as advertisable (the current default)
- Update elastic to have e.g. 'coordinator-http' and 'coordinator-transport' instead of just 'http' and 'transport' in its endpoints names, fixing its current conflicting endpoint names. Note that actual hostnames are unaffected. See endpoints example in updated elastic docs.

### CODE REVIEW

**Comment by nickbp** [comment]
Do not merge yet. Despite the status, fixes are still needed for helloworld. See prior commit where helloworld had failed (and then not re-run)

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java [hunk 1]
```diff
@@ -108,7 +111,43 @@ private static void verifyDistinctDiscoveryPrefixes(Collection<RawPod> rawPods)
...
+            }
+        }
+        if (!portNameDuplicates.isEmpty()) {
+            //TODO(nickbp): Only check this when ports are flagged with 'advertise: true' (the current default)
+            throw new IllegalArgumentException(String.format(
+                    "Duplicate port/endpoint names across different tasks: %s", portNameDuplicates));
+        }
+    }
+
+    private static <T> void collectKeys(Map<String, T> map, Set<String> seenNames, Set<String> duplicates) {
```
Comment (elezar) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java]: A style comment from my side. This method is a utility method, but mentions ports, names, and keys. We could consider changing this to:
```
private static <T> void collectKeys(Map<String, T> map, Set<String> uniques, Set<String> duplicates) {
  if (map == null) {
    return;
  }
  for (String key : map.keySet()) {
    if (!uniques.add(key)) {
      dupplicates.add(key);
    }
  }
}
```

Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java]: Forgot to generify the variable names when generifying the function. Will do so in the following PR as this one is intended to make the upcoming cut

File: docs/pages/reference/yaml-reference.md [hunk 2]
```diff
@@ -230,7 +230,7 @@ This documentation effectively reflects the Java object tree under [RawServiceSp
 
     * `ports`
 
-      The ports which your service will be using to accept incoming connections. Each port is given a unique name as follows:
+      The ports which your service will be using to accept incoming connections. Each port is given a unique name across the service, and this name is what's used to advertise that port in the `endpoints` listing. Ports may be defined a task as follows:
```
Comment (benclarkwood) [inline docs/pages/reference/yaml-reference.md]: Follow on PR: "defined in a task as..."

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java [hunk 3]
```diff
@@ -117,7 +156,7 @@ private static ReadinessCheckSpec from(RawReadinessCheck rawReadinessCheck) {
                 .build();
     }
 
-    private static DiscoverySpec from(RawDiscovery rawDiscovery) {
+    private static DiscoverySpec convertDiscovery(RawDiscovery rawDiscovery) {
```
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java]: I applaud this renaming 👍  🥇

**Comment by benclarkwood** [comment]
Verified checks.

**Pull request merged**